### PR TITLE
Arrangement Data Model 

### PR DESCRIPTION
**Pull request by Jeffchiucp**
Hi Team.

Please take a look and review the Arrangement Model (MongoDB), using JSON object that includes Snapshot, collection, item for each arrangement:

Snapshot is a version of arrangement.
Snapshots is a list, and we can access different snapshots through index of list.
User will be able to create snapshots for different arrangement.

image This source code is similar to our current NodeJS API: https://github.com/arrangement-io/arrangespace-server/blob/master/api/arrangement_api.js

What is snapshotContainer is also explained in here: https://github.com/arrangement-io/arrangethat-io-server/pull/31/

### CODE REVIEW

**Comment by Jeffchiucp** [comment]
this is actually a working progress still, since I have not implemented the RESTful API, but just connect the Model. But appreciate your feedback.

File: src/main/java/com/arrangespace/model/Arrangement.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+@Document(collection = "arrangement")
+@Getter
+@Setter
+@Data
+public class Arrangement {
+
+	@Id
+	private String id;
+	private List<Container> containers = new ArrayList<>();
```
Comment (deonjo) [inline src/main/java/com/arrangespace/model/Arrangement.java]: this is going to be an object/dictionary now.

File: src/main/java/com/arrangespace/model/Arrangement.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	private boolean is_deleted;
+	private List<Item> items = new ArrayList<>();
```
Comment (deonjo) [inline src/main/java/com/arrangespace/model/Arrangement.java]: Same here, this is the perf change

Comment (Jeffchiucp) [reply src/main/java/com/arrangespace/model/Arrangement.java]: so what does it looks like as Object/ Dictionary?

Comment (Jeffchiucp) [reply src/main/java/com/arrangespace/model/Arrangement.java]: need to update the data schema to be like the live website:
https://arrange.space/arrangement/aHOMD0LAX

Comment (natyconnor) [reply src/main/java/com/arrangespace/model/Arrangement.java]: In Java that means a Map...

Comment (Jeffchiucp) [reply src/main/java/com/arrangespace/model/Arrangement.java]: i see. I will change the data Model later

Comment (Jeffchiucp) [reply src/main/java/com/arrangespace/model/Arrangement.java]: in this PR# #5 , I will make those changes

**Comment by Jeffchiucp** [comment]
@deonjo can you take a look of the RESTful routes that we will use

**Comment by jeff20009** [comment]
here's the controller we have so far
Screen Shot 2019-10-04 at 3 22 24 PM

File: src/main/java/com/arrangespace/model/Arrangement.java (hunk 1 above)
Comment (deonjo) [reply src/main/java/com/arrangespace/model/Arrangement.java]: And here

File: src/main/java/com/arrangespace/model/Arrangement.java (hunk 2 above)
Comment (deonjo) [reply src/main/java/com/arrangespace/model/Arrangement.java]: I don’t see the change only list still

File: src/main/java/com/arrangespace/model/Arrangement.java (hunk 1 above)
Comment (Jeffchiucp) [reply src/main/java/com/arrangespace/model/Arrangement.java]: i updated and added the new class for each object

File: src/main/java/com/arrangespace/model/Arrangement.java (hunk 2 above)
Comment (Jeffchiucp) [reply src/main/java/com/arrangespace/model/Arrangement.java]: hi. i updated and added the new class for each object

**Comment by Jeffchiucp** [comment]
@deonjo  appreciate your feedback.

**Comment by Jeffchiucp** [comment]
I will make changes to this PR sometimes this week after busyness

File: src/main/java/com/arrangespace/model/Arrangement.java (hunk 2 above)
Comment (Jeffchiucp) [reply src/main/java/com/arrangespace/model/Arrangement.java]: hi. I made the change to the data model based on this https://gist.github.com/JeffChiu2009/2dad93bf307e13df39db05d8a0caa24a

**Comment by Jeffchiucp** [comment]
here's the updated data model type : https://gist.github.com/JeffChiu2009/2dad93bf307e13df39db05d8a0caa24a

This reflects the frontend schema https://github.com/arrangement-io/arrange-that-client/blob/master/src/schema/arrangementSchema.json

File: src/main/java/com/arrangespace/model/Arrangement.java (hunk 1 above)
Comment (Jeffchiucp) [reply src/main/java/com/arrangespace/model/Arrangement.java]: This reflects the frontend schema https://github.com/arrangement-io/arrange-that-client/blob/master/src/schema/arrangementSchema.json

**Comment by Jeffchiucp** [comment]
@gabekoo @deonjo 
any thought about the Arrangement API

**Comment by deonjo** [comment]
approved!

**Review by deonjo — APPROVED** [review]

**Comment by Jeffchiucp** [comment]
i added itms array support.

**Pull request merged**